### PR TITLE
226 image resize

### LINKED ISSUE #226: Video resolution does not get changed as per config file
```
2021-06-03T10:28:20 peekingduck.pipeline.nodes.input.utils.preprocess WARNING:Unable to change width of video frame to 1280, current width: 1000!
2021-06-03T10:28:20 peekingduck.pipeline.nodes.input.utils.preprocess WARNING:Unable to change height of video frame to 720, current height: 664!
```

### PR DESCRIPTION
**Pull request by diwei-tan**
Deprecate current stream.set_res as opencv uses the webcam's supported resolutions. As such, changing the source resolution is not a viable option. Instead, added a method to resize the image after reading if necessary.

Closes #226 .

### CODE REVIEW

File: peekingduck/pipeline/nodes/input/utils/preprocess.py [hunk 1]
```diff
@@ -52,3 +52,19 @@ def mirror(frame: np.array) -> np.array:
     Mirrors a video frame.
     '''
     return cv2.flip(frame, 1)
+
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/utils/preprocess.py:55]: If `set_res` is no longer used, this function can be removed.

File: peekingduck/pipeline/nodes/input/recorded.py [hunk 2]
```diff
@@ -27,7 +28,7 @@ def __init__(self, config: Dict[str, Any]) -> None:
         super().__init__(config, node_path=__name__)
         self._allowed_extensions = ["jpg", "jpeg", "png", "mp4", "avi", "mov"]
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/recorded.py:29]: Add ".mkv" extension as one of our clients recently provided videos in this format.

File: peekingduck/pipeline/nodes/input/recorded.py [hunk 3]
```diff
@@ -27,7 +28,7 @@ def __init__(self, config: Dict[str, Any]) -> None:
...
         input_dir = config['input_dir']
-        self._resolution = config['resolution']
+        self.resize_info = config['resize']
         self._mirror_image = config['mirror_image']
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/recorded.py:33]: Possible to log the old resolution here (always) and log the resized resolution (if resized)? I find this useful in past deployments.

File: peekingduck/pipeline/nodes/input/live.py [hunk 4]
```diff
@@ -26,15 +27,19 @@ class Node(AbstractNode):
     def __init__(self, config: Dict[str, Any]) -> None:
         super().__init__(config, node_path=__name__)
 
-        resolution = config['resolution']
+        self.resize_info = config['resize']
         input_source = config['input_source']
         mirror_image = config['mirror_image']
```
Comment (leeping-ng) [inline peekingduck/pipeline/nodes/input/live.py:32]: Possible to log the old resolution here (always) and log the resized resolution (if resized)? I find this useful in past deployments.

Comment (diwei-tan) [reply peekingduck/pipeline/nodes/input/live.py:32]: Added portion to log info the two resolutions

File: peekingduck/pipeline/nodes/input/recorded.py (hunk 2 above)
Comment (diwei-tan) [reply peekingduck/pipeline/nodes/input/recorded.py:29]: added.

File: peekingduck/pipeline/nodes/input/recorded.py (hunk 3 above)
Comment (diwei-tan) [reply peekingduck/pipeline/nodes/input/recorded.py:33]: Added logging. Pylint returns failure for the logs behind too similar... so I changed the wording a little.

File: peekingduck/pipeline/nodes/input/utils/preprocess.py (hunk 1 above)
Comment (diwei-tan) [reply peekingduck/pipeline/nodes/input/utils/preprocess.py:40]: removed.

**Pull request merged**